examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,585 @@
+          bandstopFilter->process(N, out.samples);
+          diagramCalculator.addFilter(bd.id, bandstopFilter);
+          break;
+        case filters::lowshelf:
+          previousParams = lowshelfFilter->getParams();
+          lowshelfFilter->setParam(1, ord.value);
+          lowshelfFilter->setParam(2, cutoffFreq.value);
+          lowshelfFilter->setParam(3, gn.value);
+          toRecompute |= previousParams[1] != ord.value || previousParams[2] != cutoffFreq.value || previousParams[3] != gn.value;
+          lowshelfFilter->process(N, out.samples);
+          diagramCalculator.addFilter(bd.id, lowshelfFilter);
+          break;
+        case filters::highshelf:
+          previousParams = highshelfFilter->getParams();
+          highshelfFilter->setParam(1, ord.value);
+          highshelfFilter->setParam(2, cutoffFreq.value);
+          highshelfFilter->setParam(3, gn.value);
+          toRecompute |= previousParams[1] != ord.value || previousParams[2] != cutoffFreq.value || previousParams[3] != gn.value;
+          highshelfFilter->process(N, out.samples);
+          diagramCalculator.addFilter(bd.id, highshelfFilter);
+          break;
+        case filters::bandshelf:
+          previousParams = bandshelfFilter->getParams();
+          bandshelfFilter->setParam(1, ord.value);
+          bandshelfFilter->setParam(2, centerFreq.value);
+          bandshelfFilter->setParam(3, freqBdWidth.value);
+          bandshelfFilter->setParam(4, gn.value);
+          bandshelfFilter->setParam(5, ripple.value);
+          toRecompute |= previousParams[1] != ord.value || previousParams[2] != centerFreq.value || previousParams[3] != freqBdWidth.value || previousParams[4] != gn.value || previousParams[5] != ripple.value;
+          bandshelfFilter->process(N, out.samples);
+          diagramCalculator.addFilter(bd.id, bandshelfFilter);
+          break;
+        default:
+          break;
+      }
+    }
+    else
+    {
+      toRecompute |= diagramCalculator.removeFilter(bd.id);
+    }
+  }
+
+  void operator()(int N)
+  {
+    auto& in = inputs.audio;
+    auto& out = outputs.audio;
+
+    auto& channels = in.channels;
+
+    //  Since DSPFilters functions modify the samples in-place,
+    //  we first copy the input into the output and then process the output samples directly.
+
+    for (auto i = 0; i < channels; ++i)
+    {
+      auto& in_samp = in.samples[i];
+      auto& out_samp = out.samples[i];
+      for (auto j = 0; j < N; ++j)
+      {
+        out_samp[j] = in_samp[j];
+      }
+    }
+
+    std::apply(
+        [&](auto&&... args)
+        {
+          (applyBand(
+               N,
+               std::get<0>(args),
+               std::get<1>(args),
+               std::get<2>(args),
+               std::get<3>(args),
+               std::get<4>(args),
+               std::get<5>(args),
+               std::get<6>(args),
+               std::get<7>(args)),
+           ...);
+        },
+        std::make_tuple(
```

Comment:
I wonder if this tuple shouldn't be put inside the class directly to be able to reuse it. idk what's more efficient honestly.